app/src/state.rs

```diff
@@ -262,7 +273,9 @@ impl State {
         let round = parts.round;
 
         // Get the expected proposer for this height and round
-        let validator_set = self.get_validator_set(); // TODO: Should pass height as a parameter
+        let validator_set = self
+            .get_validator_set(height)
+            .ok_or(ProposalValidationError::ValidatorSetNotFound { height })?;
```

Comment:
Will we have a scenario where we are asked to use a validator set from the past ? I think at the moment this is not the case. So I am not sure whether we need this in the first place and if we do ,should we extend it to query here the smart contract to get the data.

That being said, maybe it is safer to have this error here in case there is a corner case so we can catch things.

Reply:
Given the current logic I like this solution better. Please add a changelog entry and if the tests pass feel free to merge it.